File: dsa/week(4)/Mult_Inher.js
class Student {
    constructor() {
        this.name = "";
    }

    print() {
        console.log("Student");
    }
}

class Teacher {
    constructor() {
        this.name = "";
        this.age = "";
    }

    print() {
        console.log("Teacher");
    }
}

class TA extends Teacher {
    constructor() {
        super(); 
        this.studentName = "";
    }
    print() {
        console.log("TA");
    }
}


const a = new TA();
new Student().print();


a.print();

// Accessing Teacher's properties and method
a.name = "abcd";
a.print(); // Print TA's overridden function
new Teacher().print(); // Explicitly accessing Teacher's print function

File: dsa/week(4)/Mult_Inher.py
class Student:
    def __init__(self):
        self.name = ""

    def print(self):
        print("Student")


class Teacher:
    def __init__(self):
        self.name = ""
        self.age = ""

    def print(self):
        print("Teacher")


class TA(Teacher, Student):  # Inheriting from both Teacher and Student
    def print(self):
        print("TA")


# Main Logic
a = TA()

# Accessing the Student's print function
Student().print()

# Accessing the TA's print function
a.print()

# Accessing Teacher's properties and method
a.name = "abcd"
Teacher().print()
